src/librbd/cache/ReplicatedWriteLog.h

```diff
@@ -0,0 +1,501 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CACHE_REPLICATED_WRITE_LOG
+#define CEPH_LIBRBD_CACHE_REPLICATED_WRITE_LOG
+
+//#if defined(HAVE_PMEM)
+#include <libpmemobj.h>
+//#endif
+#include "common/RWLock.h"
+#include "librbd/cache/ImageCache.h"
+#include "librbd/cache/FileImageCache.h"
+#include "librbd/Utils.h"
+#include "librbd/cache/BlockGuard.h"
+#include "librbd/cache/ImageWriteback.h"
+#include "librbd/cache/file/Policy.h"
+#include "librbd/BlockGuard.h"
+#include <functional>
+#include <boost/intrusive/list.hpp>
+#include <boost/intrusive/set.hpp>
+#include <deque>
+#include <list>
+#include "common/Finisher.h"
+#include "include/assert.h"
+
+namespace librbd {
+
+struct ImageCtx;
+
+namespace cache {
+
+static const uint32_t MIN_WRITE_SIZE = 1;
+static const uint32_t BLOCK_SIZE = MIN_WRITE_SIZE;
```

Comment:
Although RBD is technically byte-addressable for IO, this seems like high overhead to support 1 byte block sizes within the cache. It seems like we should have a sane minimum but, longer term, support a non-hardcoded cache block size. I presume this was just made to support our unit tests that perform non-aligned IO?

Reply:
Right, I only lowered this below 512 because of the unit tests. The minimum allocation size is still 512, and probably shouldn't be smaller.

The block size here (and all the image extent / block extent conversion) is kind of a relic from when I thought I'd need bitmaps of sectors for something. Now I don't think I will. The API doesn't care about block size, and now I'm not sure why RWL should.

Reply:
Good point -- it could certainly remove a good chunk of code if you didn't need to be block aligned.